MobiFlight/JoystickManager.cs

```diff
@@ -77,12 +78,14 @@ public void Connect(IntPtr Handle)
                 if (!IsSupportedDeviceType(d)) continue;
 
                 MobiFlight.Joystick js;
-                if (d.InstanceName == "Bravo Throttle Quadrant")
-                {
-                    js = new Joysticks.HoneycombBravo(new SharpDX.DirectInput.Joystick(di, d.InstanceGuid));
-                } else if (d.InstanceName == "Saitek Aviator Stick")
+
+                // Look up the custom device definition file for the joystick. If it exists then a LabeledJoystick
+                // can get created with nicely named inputs and outputs.
+                var definition = HidDefinitions.GetHidByInstanceName(d.InstanceName);
+
+                if (definition != null)
                 {
-                    js = new Joysticks.SaitekAviatorStick(new SharpDX.DirectInput.Joystick(di, d.InstanceGuid));
+                    js = new Joysticks.LabeledJoystick(new SharpDX.DirectInput.Joystick(di, d.InstanceGuid), definition);
```

Comment:
The meat of the change in JoystickManager. Look up a matching board definition by instance name and if it's found use that to create a `LabeledJoystick` object. If no definition is found then fall back to making a generic `Joystick`.